pkg/controller/proposal/controller.go

```diff
@@ -456,10 +457,19 @@ func (r *Reconciler) reconcileCommit(ctx context.Context, proposal *configapi.Pr
 			if config.Values == nil {
 				config.Values = make(map[string]*configapi.PathValue)
 			}
-			for path, changeValue := range changeValues {
-				_, _ = applyChangeToConfig(config.Values, path, changeValue)
-			}
+			// ToDo - remove before merging
+			log.Debugf("Commit phase (461) - values are:\n%v", config.Values)
+			log.Debugf("Commit phase (462) - Following paths are required to be updated:\n%v", changeValues)
+			updatedChangeValues := controllerutils.AddDeleteChildren(changeValues, config.Values)
```

Comment:
I think we agreed of handing this behind a variable, maybe not all targets want a cascading delete to happen. @adibrastegarnia @SeanCondon am I correct ?

Reply:
I think cascading delete is something generic. That means when you delete an intermediate node in the config tree, it makes sense the whole subtree gets deleted. If the user doesn't want to delete the whole subtree then the user should just delete leaf values or just the paths that it cares about. I don't think enabling/disabling per target makes sense in general.